test/shared/src/main/scala/zio/test/DefaultTestReporter.scala

```diff
@@ -109,6 +110,36 @@ object DefaultTestReporter {
     } yield ()
   }
 
+  private def renderTimed[L, E, S](executedSpec: ExecutedSpec[L, E, S]): UIO[Option[String]] = {
+    val results = executedSpec.fold[UIO[Vector[(L, Duration)]]] {
+      case Spec.SuiteCase(_, executedSpecs, _) =>
+        executedSpecs.flatMap(UIO.collectAll(_).map(_.foldLeft(Vector.empty[(L, Duration)])(_ ++ _)))
+      case Spec.TestCase(label, test) =>
+        test.map {
+          case (_, annotationMap) =>
+            val d = annotationMap.get(TestAnnotation.Timing)
```

Comment:
We should think how to pull this logic out of here and push it higher.

Reply:
Yes. I think once we have `TestAnnotationRenderer`, `TestReporter` can take a `List[TestAnnotationRenderer]` as input, and the `DefaultTestReporter` in particular can have a set of default renderers in addition to whichever ones are passed to it. Then the `DefaultTestReporter` can traverse the executed spec, report all annotations it has renderers for, and either ignore or report an error for any others.